pkg/apis/resource/validation/validation.go

```diff
@@ -176,9 +176,10 @@ func validateDeviceRequest(request resource.DeviceRequest, fldPath *field.Path,
 	numDeviceRequestType := 0
 	if len(request.FirstAvailable) > 0 {
 		numDeviceRequestType++
-		allErrs = append(allErrs, validateSet(request.FirstAvailable, resource.FirstAvailableDeviceRequestMaxSize,
-			func(subRequest resource.DeviceSubRequest, fldPath *field.Path) field.ErrorList {
-				return validateDeviceSubRequest(subRequest, fldPath, stored)
+		allErrs = append(allErrs, validateSetWithIndex(request.FirstAvailable, resource.FirstAvailableDeviceRequestMaxSize,
+			func(subRequest resource.DeviceSubRequest, fldPath *field.Path, index int) field.ErrorList {
+				lastInList := index == len(request.FirstAvailable)-1
+				return validateDeviceSubRequest(subRequest, fldPath, stored, lastInList)
```

Comment:
It's worth calling out that this change of validation is okay because it only affects an alpha-level field.